server/routes/user.js

```diff
@@ -0,0 +1,81 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const multer = require('multer');
+const { GridFsStorage } = require('multer-gridfs-storage');
+const Grid = require('gridfs-stream');
+
+// Setup MongoDB connection
+const mongoURI = process.env.MONGO_URI;
+const conn = mongoose.createConnection(mongoURI, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true
+});
+
+// Initialize GridFS
+let gfs;
+conn.once('open', () => {
+    gfs = Grid(conn.db, mongoose.mongo);
+    gfs.collection('avatars');
+});
+
+// Storage engine for multer and GridFS
+const storage = new GridFsStorage({
+    url: mongoURI,
+    file: (req, file) => {
+        return {
+            filename: `${req.body.username}-avatar-${Date.now()}`,
+            bucketName: 'avatars',
+        };
+    },
+});
+const upload = multer({ storage });
+
+// Define user schema with bio field
+const userSchema = new mongoose.Schema({
+    username: { type: String, required: true, unique: true },
+    bio: { type: String },
+});
+
+const User = mongoose.model('User', userSchema);
+
+// Initialize router
+const router = express.Router();
+
+// Save bio endpoint
+router.post('/user/bio', async (req, res) => {
+    const { username, bio } = req.body;
+
+    if (!username || !bio) {
+        return res.status(400).json({ message: 'Username and bio are required.' });
+    }
+
+    try {
+        let user = await User.findOne({ username });
+        if (!user) {
```

Comment:
If the user doesn't exist we probably want to return an error and not create a new user